packages/uni_app/lib/model/utils/day_of_week.dart

```diff
@@ -8,6 +8,28 @@ enum DayOfWeek {
   sunday
 }
 
+DayOfWeek parseDateTime(DateTime dateTime) {
+  final weekDay = dateTime.weekday;
+  switch (weekDay) {
+    case DateTime.monday:
+      return DayOfWeek.monday;
+    case DateTime.tuesday:
+      return DayOfWeek.tuesday;
+    case DateTime.wednesday:
+      return DayOfWeek.wednesday;
+    case DateTime.thursday:
+      return DayOfWeek.thursday;
+    case DateTime.friday:
+      return DayOfWeek.friday;
+    case DateTime.saturday:
+      return DayOfWeek.saturday;
+    case DateTime.sunday:
+      return DayOfWeek.sunday;
+    default:
+      throw Exception('Invalid day of week');
+  }
+}
+
```

Comment:
This is really weird, but I understand why this had to be done :laughing: 
Maybe we can use `WeekdayMapper` for this? 

```suggestion
DayOfWeek parseDateTime(DateTime dateTime) {
  final index = WeekdayMapper.fromDartToIndex.map(dateTime.weekday);
  return DayOfWeek.values[index];
}

```

Don't forget to update the imports to include `WeekdayMapper`!

Reply:
I didn’t know that existed, hahaha! I even started changing the entire codebase to stop using DayOfWeek and switch to DateTime.weekday, but it would have required changing 50+ files, so I decided not to go through with it 😅.

Reply:
Although I understand the motivation, it is a decision that must be made carefully... DateTime's weekday is an int, which allows 2+ billion values and only 7 of them are correct, while DayOfWeek only allows 7 values, and all of them are correct.
Imo, it's best to add an extension to the `int` type to convert it to DayOfWeek and vice-versa!